docs: add howto about emergency disable (#588)
kubewarden/docs: docs/howtos/emergency-disable.md

---
sidebar_label: Emergency disable
sidebar_position: 31
title: Emergency disable
description: How to temporarily disable Kubewarden policies during emergencies
keywords: [ClusterAdmissionPolicies, AdmissionPolicies, configuration]
doc-persona: [kubewarden-operator]
doc-type: [howto]
doc-topic: [operator-manual]
---

<head>
  <link rel="canonical" href="https://docs.kubewarden.io/howtos/emergency-disable"/>
</head>

During an emergency situation, operation teams might need to perform actions that can be blocked by Kubewarden.

This document explains how it's possible to temporarily disable Kubewarden to allow operation teams to address cluster failures.

## Disable Kubewarden

First, you have to stop the Kubewarden controller. This can be done by scaling its Deployment to zero:

```console
kubectl scale deployment kubewarden-controller \
  --replicas=0 \
  -n kubewarden
```

:::tip

> Ensure you put the name of the Namespace where the Kubewarden stack has been deployed.

Reviewer: ```suggestion
Ensure you use the name of the Namespace where the Kubewarden stack is deployed.
```

The previous example assumes the stack has been deployed inside of the `kubewarden` Namespace.
:::

Next, delete all the `ValidatingWebhookConfigurations` and `MutatingWebhookConfigurations`

> that have been created by Kubewarden:

Reviewer: ```suggestion
created by Kubewarden:
```

```console
# Delete all the ValidatingWebhookConfiguration created by Kubewarden
kubectl delete validatingwebhookconfigurations \
  -l app.kubernetes.io/part-of=kubewarden

# Delete all the MutatingWebhookConfiguration created by Kubewarden
kubectl delete mutatingwebhookconfigurations \
  -l app.kubernetes.io/part-of=kubewarden
```

## Restore Kubewarden

> Once the emergency is over, the Kubewarden stack can be restored by bringing back
> the Kubewarden controller:

Reviewer: ```suggestion
When the emergency is over, restore the Kubewarden stack by bringing back
the Kubewarden controller:
```

```console
kubectl scale deployment kubewarden-controller \
  --replicas=1 \
  -n kubewarden
```

:::tip

> - Ensure you put the name of the Namespace where the Kubewarden stack has been deployed.

Reviewer: ```suggestion
- Ensure you use the name of the Namespace where the Kubewarden stack is deployed.
```

>   The previous example assumes the stack has been deployed inside of the `kubewarden` Namespace.

Reviewer: ```suggestion
  Here, the stack has been deployed in the `kubewarden` Namespace.
```

> - Ensure you scale the controller back to its original value. The previous example assumes
>   there's only one replica of the controller running.

Reviewer: ```suggestion
- Ensure you scale the controller back to its original value. This example assumes
  there's only one replica of the controller running.
```

  :::

> Once the controller is running, it will take care of reconciling the deployed
> policies. The `ValidatingWebhookConfiguration` and `MutatingWebhookConfiguration` resources
> previously deleted will be recreated, thus enforcing the policies.

Reviewer: ```suggestion
Once the controller is running, it reconciles the deployed
policies. The `ValidatingWebhookConfiguration` and `MutatingWebhookConfiguration` resources,
previously deleted, are recreated, thus enforcing the policies.
```
